rhel/vex/parser.go

```diff
@@ -0,0 +1,450 @@
+			// containers have no say in the kernel.
+			continue
+		}
+		if purl.Type != packageurl.TypeRPM || purl.Namespace != "redhat" {
+			// Just ingest advisories that are Red Hat RPMs, this will
+			// probably change down the line when we consolidate updaters.
+			continue
+		}
+
+		fixedIn := epochVersion(&purl)
+		vulnKey := createPackageKey(rel.RelatesToProductRef, purl.Name, fixedIn)
+		arch := purl.Qualifiers.Map()["arch"]
+		if vuln, ok := c.lookupVulnerability(vulnKey, protoVulnFunc); ok && arch != "" {
+			// We've already found this package, just append the arch
+			vuln.Package.Arch = vuln.Package.Arch + "|" + arch
+		} else {
+			vuln.FixedInVersion = fixedIn
+			vuln.Package = &claircore.Package{
+				Name: purl.Name,
+				Kind: claircore.BINARY,
+			}
+
+			if arch != "" {
+				vuln.Package.Arch = arch
+				vuln.ArchOperation = claircore.OpPatternMatch
+			}
+
+			wfn, err := cpe.Unbind(cpeHelper)
+			if err != nil {
+				return nil, fmt.Errorf("could not unbind cpe: %s %w", cpeHelper, err)
+			}
+			vuln.Repo = c.rc.Get(wfn)
+			// Find remediations and add RHSA URL to links
+			rem := c.c.FindRemediation(pc)
+			if rem != nil {
+				vuln.Links = vuln.Links + " " + rem.URL
+			}
+			if sc := c.c.FindScore(pc); sc != nil {
+				vuln.NormalizedSeverity, vuln.Severity, err = CVSSVectorFromScore(sc)
+				if err != nil {
+					return nil, fmt.Errorf("could not parse CVSS score: %w, file: %s", err, c.vulnLink)
+				}
+			}
+		}
+	}
+	out := make([]*claircore.Vulnerability, len(c.fixedVulns))
+	for i := range c.fixedVulns {
+		out[i] = &c.fixedVulns[i]
+	}
+	return out, nil
+}
+
+func CVSSVectorFromScore(sc *csaf.Score) (sev claircore.Severity, vec string, err error) {
+	switch {
+	case sc.CVSSV4 != nil:
+		var c cvss.V4
+		c, err = cvss.ParseV4(sc.CVSSV4.VectorString)
+		if err != nil {
+			err = fmt.Errorf("could not parse CVSSv4 vector string %w", err)
+			return
+		}
+		sev = CVSS2ClaircoreSeverity(cvss.QualitativeScore[cvss.V4Metric](&c))
+		vec = sc.CVSSV4.VectorString
+		return
+	case sc.CVSSV3 != nil:
+		var c cvss.V3
+		c, err = cvss.ParseV3(sc.CVSSV3.VectorString)
+		if err != nil {
+			err = fmt.Errorf("could not parse CVSSv3 vector string %w", err)
+			return
+		}
+		sev = CVSS2ClaircoreSeverity(cvss.QualitativeScore[cvss.V3Metric](&c))
+		vec = sc.CVSSV3.VectorString
+		return
+	case sc.CVSSV2 != nil:
+		var c cvss.V2
+		c, err = cvss.ParseV2(sc.CVSSV2.VectorString)
+		if err != nil {
+			err = fmt.Errorf("could not parse CVSSv4 vector string %w", err)
+			return
+		}
+		sev = CVSS2ClaircoreSeverity(cvss.QualitativeScore[cvss.V2Metric](&c))
+		vec = sc.CVSSV2.VectorString
+		return
+	default:
+		err = errors.New("could not find a valid CVSS object")
+	}
+	return
+}
+
+// CVSS2ClaircoreSeverity returns a claircore.Severity given a cvss.Qualitative
+// It will panic if asked to reason with an unknown Qualitative score.
+func CVSS2ClaircoreSeverity(q cvss.Qualitative) claircore.Severity {
```

Comment:
why the 2? I thought this was specifically for CVSS 2.0 at first

Reply:
Ditto, I think this a bad name